xdsl/dialects/linalg.py

```diff
@@ -96,8 +96,8 @@ class Generic(IRDLOperation):
     # Trait attributes
     indexing_maps: ArrayAttr[AffineMapAttr] = prop_def(ArrayAttr[AffineMapAttr])
     iterator_types: ArrayAttr[IteratorTypeAttr] = prop_def(ArrayAttr[IteratorTypeAttr])
-    doc: StringAttr | None = opt_attr_def(StringAttr)
-    library_call: StringAttr | None = opt_attr_def(StringAttr)
+    doc: StringAttr | None = opt_prop_def(StringAttr)
```

Comment:
do you think it would be too much of a detour to implement default properties before merging this?

Reply:
I really want to - and soon! Though this is actually a sensitive fix: MLIR interop is currently broken without it. Can I have your approval to merge this frustrating `opt_prop_def` so it's functional first? We'll get rid of those eventually!

Reply:
I'm now wondering whether opt_attr_def will work for you in the short term? If not then let's merge like this.